api/v1alpha2/awsmachine_conversion.go

```diff
@@ -23,48 +23,50 @@ import (
 )
 
 // ConvertTo converts this AWSMachine to the Hub version (v1alpha3).
-func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
+func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1alpha3.AWSMachine)
-	return Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil)
+	if err := Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Any particular reason for this change? It is definitely a bit more explicit about the error handling, but it doesn't actually change any behavior and doesn't match with the other common usage in this file.

Reply:
I was initially going to add the restore logic, then I took it away. I can revert it back if you prefer, this is just first step

Reply:
Let's revert to avoid conflicts with the #1498